src/linear_combination.jl

```diff
@@ -0,0 +1,59 @@
+"""
+    LinearCombination <: Transform
+
+Calculate the linear combination using the column weights passed in.
+"""
+struct LinearCombination <: Transform
+    col_weights::Vector{Real}
+end
+
+function _check_dimensions_match(LC::LinearCombination, num_cols)
+    num_weights = length(LC.col_weights)
+    if num_cols != num_weights
+        throw(DimensionMismatch(
+            "Number of cols ($num_cols) doesn't match number of weights ($num_weights)"
+        ))
+    end
+end
+
+_sum_row(row, col_weights) = sum(map(*, row, col_weights))
+
+"""
+    apply(x, LC::LinearCombination; cols=Colon())
+
+Applies the [`LinearCombination`](@ref) to each of the specified columns in `x`.
+If no `cols` are specified, then the [`LinearCombination`](@ref) is applied to all columns.
+"""
+function apply(x::AbstractVector, LC::LinearCombination; cols=Colon())
+    # Treat each element as it's own column
+    # Error if dimensions don't match
+    num_cols = cols isa Colon ? length(x) : length(cols)
+    _check_dimensions_match(LC, num_cols)
+
+    return [_sum_row(x[cols], LC.col_weights)]
+end
+
+function apply(x::AbstractArray, LC::LinearCombination; cols=Colon())
+    # Error if dimensions don't match
+    num_cols = cols isa Colon ? size(x, 2) : length(cols)
+    _check_dimensions_match(LC, num_cols)
+
+    return [_sum_row(row[cols], LC.col_weights) for row in eachrow(x)]
```

Comment:
```suggestion
    return [_sum_row(row[cols], LC.col_weights) for row in eachslice(x; dims=dims)]
```

here is where `dims` could come into play